src/main/java/com/example/dog/controller/DogController.java

```diff
@@ -11,15 +11,15 @@
 @RestController
 @RequestMapping(path="/dog", produces="application/json")
 public class DogController {
-    private final DogService service;
+    private final TransactionalDogService transactionalDogService;
```

Comment:
Nope. That's the whole point of Proxy/Decorator - you should work with the interface so that the rest of the code doesn't know that you replaced the original object with a proxy. They should be undistinguishable.